ONSdigital/ssdc-rm-support-tool
# 2450 Case search enhancements

# Motivation and Context
We want to be able to search across multiple sample fields at once across an entire survey and filter the results on certain case properties.

# What has changed
- Add search in sample data endpoint to backend
- Add find by case ref endpoint to backend
- Add find by qid endpoint to backend
- Add new case search page to UI
- Survey be survey access control

# How to test?
1. Start up docker dev
2. Stop the supporttool container `docker stop supporttool`
3. Checkout this branch of support tool
4. Start up the support tool backend java service locally in the IDE
5. Run the UI server with `make run-dev-ui`
6. Navigate to the support tool UI on `localhost:3000`
7. Open the details for a survey, you should now see a search cases button, allowing you to access and play with the new search functionality

# Links
https://trello.com/c/RlCtolCu/2450-ops-tool-case-search-13

# Screenshots (if appropriate):
![Screenshot 2021-07-16 at 14 10 12](https://user-images.githubusercontent.com/35268982/125953563-89487df0-476f-4ed7-838f-32e79809684f.png)
